components/auth/RegisterAgreement.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import palette from '../../lib/styles/palette';
+import { CheckboxItem } from '.';
+import { registerFormItems } from '../../lib/texts/texts';
+import { RoundButton } from '..';
+
+type AgreementProps = {
+  toPageNext: Function;
+  registerForm: object;
+  setRegisterForm: Function;
+};
+
+export const RegisterAgreement = ({
+  toPageNext,
+  registerForm,
+  setRegisterForm,
+}: AgreementProps) => {
+  const [isShowAll, setIsShowAll] = useState(false);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [isCheckedAll, setIsCheckedAll] = useState(false);
+  // BasicTerm, PersonalInfo, SMS, Email
+  const [checkedList, setCheckedLists] = useState([false, false, false, false]);
+
+  const showAll = () => {
+    setIsShowAll(true);
+  };
+
+  const handleCheckAll = () => {
+    setCheckedLists([
+      !isCheckedAll,
+      !isCheckedAll,
+      !isCheckedAll,
+      !isCheckedAll,
+    ]);
+    setIsCheckedAll(!isCheckedAll);
+  };
+
+  const handleChecked = (index: number) => {
+    let arrayCopied = [...checkedList];
+    arrayCopied[index] = !checkedList[index];
+    setIsCheckedAll(arrayCopied.every(Boolean));
+    setCheckedLists(arrayCopied);
+  };
+
+  const onButtonClick = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+    if (!(checkedList[0] && checkedList[1])) {
+      setIsErrorShown(true);
+    } else {
+      let agreementsAdded = {
+        ...registerForm,
+        agreements: checkedList,
+      };
+      setRegisterForm(agreementsAdded);
+      toPageNext();
+    }
```

Comment:
이런 경우에는 early-return 을 하면 주로 실행되는 코드의 깊이를 얕게 만들 수 있습니다. 
```suggestion
    if (!(checkedList[0] && checkedList[1])) {
      setIsErrorShown(true);
      return;
    }

    const agreementsAdded = {
      ...registerForm,
      agreements: checkedList,
    };
    setRegisterForm(agreementsAdded);
    toPageNext();
```